### PR TITLE
new: API GET /projects/id/datasets

### PR DESCRIPTION
**Pull request by sadiqkhoja**
Created new API that returns all datasets of a given project.
Endpoint: `/v1/projects/id/datasets`
Response contains only published datasets, by published I mean the corresponding form has been published

#### What has been done to verify that this works as intended?
Integration tests

#### Why is this the best possible solution? Were any other approaches considered?
hmmm

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
None

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.
Will do it in a separate PR

#### Before submitting this PR, please make sure you have:

- [X] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [NA] verified that any code from external sources are properly credited in comments

### CODE REVIEW

File: lib/resources/projects.js [hunk 1]
```diff
@@ -32,6 +32,12 @@ module.exports = (service, endpoint) => {
         ? auth.verbsOn(project).then((verbs) => Object.assign({ verbs }, project.forApi()))
         : project))));
 
+  service.get('/projects/:id/datasets', endpoint(({ Projects, Datasets }, { auth, params, queryOptions }) =>
```
Comment (sadiqkhoja) [inline lib/resources/projects.js:35]: move this endpoint to another resource file

File: test/integration/api/projects.js [hunk 2]
```diff
@@ -1269,6 +1269,43 @@ describe('api: /projects', () => {
                 .then((o) => { o.isDefined().should.equal(false); })
             ])))))));
   });
+
+  describe('/:id/datasets GET', () => {
```
Comment (sadiqkhoja) [inline test/integration/api/projects.js:1273]: move these tests to new datasets test suites

File: lib/model/query/datasets.js [hunk 3]
```diff
@@ -50,7 +50,12 @@ const isNilOrEmpty = either(isNil, isEmpty);
 
 ////////////////////////////////////////////////////////////////////////////////
 // SQL Queries
-const _getAllByProjectId = (projectId) => sql`select ${Dataset.fieldlist} from datasets WHERE "projectId" = ${projectId}`;
+const _getAllByProjectId = (projectId) => sql`
+  SELECT DISTINCT d.* FROM datasets d 
+  JOIN dataset_defs dd ON d.id = dd."datasetId" 
+  JOIN form_defs fd ON fd.id = dd."formDefId" 
+  WHERE fd."publishedAt" IS NOT NULL AND d."projectId" = ${projectId}
```
Comment (sadiqkhoja) [inline lib/model/query/datasets.js:57]: order by `name`

**Review by ktuite — APPROVED** [review]
Go ahead and merge once things are moved to a datasets resource/tests!

Questions/todo items for later PRs
- verbs/permissions and tests for verbs
- do we want to add `createdAt` to datasets or just use implicit form publish timestamp?

**Pull request merged**